packages/adapters/subgraph/src/lib/operations/queries.ts

```diff
@@ -154,9 +159,16 @@ export const getAssetBalancesQuery = (prefix: string, router: string): string =>
     `;
 };
 
-export const getAssetBalancesAllRoutersQuery = (prefix: string): string => {
+export const getAssetBalancesRoutersQuery = (
+  prefix: string,
+  offset: number,
+  orderDirection: "asc" | "desc" = "desc",
+): string => {
   const queryString = `
-    ${prefix}_routers {
+    ${prefix}_routers(
+    skip: ${offset},
```

Comment:
We should also use the `first` param here so we can be explicit about how big the pages are.

Reply:
Changed it use first as well, with explicit page size.